cmd/pkf/internal/http/handlers/pubkey_test.go

```diff
@@ -0,0 +1,180 @@
+				"address":  "0x5602ea95540bee46d03ba335eed6f49d117eab95c8ab8b71bae2cdd1e564a761",
+				"network":  "mainnet",
+				"protocol": "ethereum",
+			},
+			http.StatusInternalServerError,
+		},
+		{
+			"500-encryption-methods-unsupported-public-key-type",
+			args{
+				func() datastore.PublicKeyStore {
+					store := datastoretest.NewMockPublicKeyStore(mockCtrl)
+					res := &datastore.PublicKey{
+						PublicKey: &unknownPublicKey{},
+						BlockHash: blockHash,
+						TxHash:    txHash,
+					}
+					store.EXPECT().GetPublicKey(gomock.Any(), "ethereum", "mainnet", address).Return(res, nil).Times(1)
+					return store
+				}(),
+			},
+			map[string]string{
+				"address":  "0xD5ab4CE3605Cd590Db609b6b5C8901fdB2ef7FE6",
+				"network":  "mainnet",
+				"protocol": "ethereum",
+			},
+			http.StatusInternalServerError,
+		},
+		{
+			"200-sofia-secp256k1",
+			args{
+				func() datastore.PublicKeyStore {
+					store := datastoretest.NewMockPublicKeyStore(mockCtrl)
+					res := &datastore.PublicKey{
+						PublicKey: secp256k1test.SofiaPublicKey,
+						BlockHash: blockHash,
+						TxHash:    txHash,
+					}
+					store.EXPECT().GetPublicKey(gomock.Any(), "ethereum", "mainnet", address).Return(res, nil).Times(1)
+					return store
+				}(),
+			},
+			map[string]string{
+				"address":  "0xD5ab4CE3605Cd590Db609b6b5C8901fdB2ef7FE6",
+				"network":  "mainnet",
+				"protocol": "ethereum",
+			},
+			http.StatusOK,
+		},
+		{
+			"200-charlotte-secp256k1",
+			args{
+				func() datastore.PublicKeyStore {
+					store := datastoretest.NewMockPublicKeyStore(mockCtrl)
+					res := &datastore.PublicKey{
+						PublicKey: secp256k1test.CharlottePublicKey,
+						BlockHash: blockHash,
+						TxHash:    txHash,
+					}
+					store.EXPECT().GetPublicKey(gomock.Any(), "ethereum", "mainnet", address).Return(res, nil).Times(1)
+					return store
+				}(),
+			},
+			map[string]string{
+				"address":  "0xD5ab4CE3605Cd590Db609b6b5C8901fdB2ef7FE6",
+				"network":  "mainnet",
+				"protocol": "ethereum",
+			},
+			http.StatusOK,
+		},
+	}
+	for _, tt := range tests {
+		testName := t.Name()
+		t.Run(tt.name, func(t *testing.T) {
+			req, _ := http.NewRequest("GET", "/", nil)
+			q := req.URL.Query()
+			for k, v := range tt.queryParams {
+				q.Add(k, v)
+			}
+			req.URL.RawQuery = q.Encode()
+
+			// We create a ResponseRecorder (which satisfies http.ResponseWriter) to record the response.
+			rr := httptest.NewRecorder()
+			handler := http.HandlerFunc(GetPublicKey(tt.args.store))
+
+			// Our handlers satisfy http.Handler, so we can call their ServeHTTP method
+			// directly and pass in our Request and ResponseRecorder.
+			handler.ServeHTTP(rr, req)
+
+			// Check the status code is what we expect.
+			if !assert.Equal(t, tt.wantStatus, rr.Code) {
+				t.Errorf("handler returned wrong status code: got %v want %v",
+					rr.Code, tt.wantStatus)
+			}
+
+			golden, err := ioutil.ReadFile(fmt.Sprintf("./testdata/%s/%s.json", testName, tt.name))
```

Comment:
Can you add `-request` to `golden, err := ioutil.ReadFile(fmt.Sprintf("./testdata/%s/request-%s.json", testName, tt.name))` then add prefix of `request-` to each test file. This will make it clearer when there requests and responses.

Reply:
Since it is th response json, I have added `response-`